nvflare/app_common/executors/model_learner_executor.py

```diff
@@ -84,7 +84,11 @@ def handle_event(self, event_type: str, fl_ctx: FLContext):
 
     def _create_learner(self, fl_ctx: FLContext):
         engine = fl_ctx.get_engine()
-        self.learner = engine.get_component(self.learner_id)
+        if isinstance(self.learner_id, str):
+            self.learner = engine.get_component(self.learner_id)
```

Comment:
@yhwen could we have a general solution for this? Maybe get_component could return the python object directly if it is not a string but a component already?

Reply:
Why do we need to change this logic in this PR?

Reply:
After discussion, we agreed to change the `engine.get_component(component_id)` to return the component directly if already a component and not a id string. This allows not having to change the logic in this file. I updated the doc string though.